### PR TITLE
docs: update code snippets and more

### LINKED ISSUE #343: Outdated code snippets
**Describe the problem**

The code snippets are mostly outdated (javascript, still using `ARK_XXX` env vars, ...) and should be replaced. Should the replacing code snippets be taken from the core `master` or `develop` branch @faustbrian?

### PR DESCRIPTION
**Pull request by dated**
## Proposed changes

Many docs, code snippets and references are outdated and still use or point to the old javascript code. Closes #343.

## Types of changes

- [x] Docs (documentation only changes)

## Checklist

- [x] I have read the [CONTRIBUTING](/guidebook/contribution-guidelines/contributing.html) documentation
- [x] I have read the [WRITING DOCUMENTATION](/guidebook/contribution-guidelines/writing-documentation.html) guidelines

### CODE REVIEW

**Comment by faustbrian** [comment]
@dated ready to be checked and merged?

**Comment by dated** [comment]
- All multi-line code snippets have been substituted with their typescript counterpart and the code blocks marked with `ts`
 - Single line codeblocks are still marked with `js`, e.g.
    ```js
    transactions.create;
    ```
 - I removed and adapted parts of the docs to my best knowledge, but this PR does not focus on content

Ready to be checked, yea. If you need more adjustments let me know.

File: docs/guidebook/core/plugins/required/core-p2p.md [hunk 1]
```diff
@@ -14,36 +14,85 @@ You can find the source code of this package at [packages/core-p2p](https://gith
...
-    userCache: {
-      expiresIn: 1000,
+```ts
+export const defaults = {
+    host: process.env.CORE_P2P_HOST || "0.0.0.0",
+    port: process.env.CORE_P2P_PORT || 4002,
+    /**
+     * The minimum peer version we expect
+     */
+    minimumVersions: [">=2.1.0", ">=2.2.0-alpha.0", ">=2.2.0-beta.0", ">=2.2.0-rc.0", ">=2.2.0-next.0"],
```
Comment (faustbrian) [inline docs/guidebook/core/plugins/required/core-p2p.md]:
```suggestion
    minimumVersions: [">=2.3.0", ">=2.3.0-next.0"],
```

**Review by faustbrian — APPROVED** [review]

**Comment by ghost** [comment]
A contributor has approved this PR. A maintainer will merge this PR shortly. If it shouldn't be merged yet, please leave a comment saying so and we'll wait.

_Thank you for your contribution!_

**Comment by ghost** [comment]
Your pull request has been merged and marked as tier 2. _It will earn you $50 USD._

**Pull request merged**